tidy3d/components/simulation.py

```diff
@@ -793,6 +793,35 @@ def _check_normalize_index(cls, val, values):
 
         return val
 
+    """ Post-init validators """
+
+    def _post_init_validators(self) -> None:
+        """Call validators taking z`self` that get run after init."""
+        self._validate_no_structures_pml()
+        self._validate_tfsf_nonuniform_grid()
+
+    def _validate_no_structures_pml(self) -> None:
+        """Ensure no structures terminate / have bounds inside of PML."""
+
+        pml_thicks = self.pml_thicknesses
+        sim_bounds = self.bounds
+        for i, structure in enumerate(self.structures):
+            geo_bounds = structure.geometry.bounds
+            for sim_bound, geo_bound, pml_thick in zip(sim_bounds, geo_bounds, pml_thicks):
+                for sim_pos, geo_pos, pml, pm_val in zip(sim_bound, geo_bound, pml_thick, (-1, 1)):
+                    sim_pos_pml = sim_pos + pm_val * pml
+                    in_pml_plus = (pm_val > 0) and (sim_pos < geo_pos <= sim_pos_pml)
+                    in_pml_mnus = (pm_val < 0) and (sim_pos > geo_pos >= sim_pos_pml)
```

Comment:
I think we should avoid this warning if it's an Absorber, since there, translational invariance is not needed.

Reply:
ok I added this check and a test